resources/packs/aws/aws_ecr.go

```diff
@@ -0,0 +1,239 @@
+	poolOfJobs.Run()
+
+	// check for errors
+	if poolOfJobs.HasErrors() {
+		return nil, poolOfJobs.GetErrors()
+	}
+	// get all the results
+	for i := range poolOfJobs.Jobs {
+		res = append(res, poolOfJobs.Jobs[i].Result.([]interface{})...)
+	}
+
+	return res, nil
+}
+
+func (e *mqlAwsEcr) getPrivateRepositories(provider *aws.Provider) []*jobpool.Job {
+	tasks := make([]*jobpool.Job, 0)
+	ctx := context.Background()
+
+	regions, err := provider.GetRegions()
+	if err != nil {
+		return []*jobpool.Job{{Err: err}} // return the error
+	}
+	for ri := range regions {
+		region := regions[ri]
+		f := func() (jobpool.JobResult, error) {
+			svc := provider.Ecr(region)
+			res := []interface{}{}
+
+			repoResp, err := svc.DescribeRepositories(ctx, &ecr.DescribeRepositoriesInput{})
+			if err != nil {
+				return nil, err
+			}
+			for i := range repoResp.Repositories {
+				r := repoResp.Repositories[i]
+				mqlRepoResource, err := e.MotorRuntime.CreateResource("aws.ecr.repository",
+					"arn", core.ToString(r.RepositoryArn),
+					"name", core.ToString(r.RepositoryName),
+					"uri", core.ToString(r.RepositoryUri),
+					"registryId", core.ToString(r.RegistryId),
+					"public", false,
+					"region", region,
+				)
+				if err != nil {
+					return nil, err
+				}
+				res = append(res, mqlRepoResource)
+			}
+
+			return jobpool.JobResult(res), nil
+		}
+		tasks = append(tasks, jobpool.NewJob(f))
+	}
+	return tasks
+}
+
+func (e *mqlAwsEcrRepository) GetImages() ([]interface{}, error) {
+	name, err := e.Name()
+	if err != nil {
+		return nil, errors.Wrap(err, "unable to parse name")
+	}
+	region, err := e.Region()
+	if err != nil {
+		return nil, errors.Wrap(err, "unable to parse region")
+	}
+	public, err := e.Public()
+	if err != nil {
+		return nil, errors.Wrap(err, "unable to parse public val")
+	}
+	at, err := awsProvider(e.MotorRuntime.Motor.Provider)
+	if err != nil {
+		return nil, err
+	}
+	ctx := context.Background()
+	mqlres := []interface{}{}
+	if public {
+		svc := at.EcrPublic(region)
+		res, err := svc.DescribeImages(ctx, &ecrpublic.DescribeImagesInput{RepositoryName: &name})
+		if err != nil {
+			return nil, err
+		}
+
+		for i := range res.ImageDetails {
+			image := res.ImageDetails[i]
+			tags := []interface{}{}
+			for i := range image.ImageTags {
+				tags = append(tags, image.ImageTags[i])
+			}
+			mqlImage, err := e.MotorRuntime.CreateResource("aws.ecr.image",
+				"digest", core.ToString(image.ImageDigest),
+				"mediaType", core.ToString(image.ImageManifestMediaType),
+				"tags", tags,
+				"registryId", core.ToString(image.RegistryId),
+				"repoName", name,
+			)
+			if err != nil {
+				return nil, err
+			}
+			mqlres = append(mqlres, mqlImage)
+		}
+	} else {
+		svc := at.Ecr(region)
+		res, err := svc.DescribeImages(ctx, &ecr.DescribeImagesInput{RepositoryName: &name})
+		if err != nil {
+			return nil, err
+		}
+		for i := range res.ImageDetails {
```

Comment:
I do not see any difference in the code, once you got the response from `DescribeImages`.
You could move this out of the `if public` to have the `for` loop only once.

Reply:
the only issue there is the types are different

Reply:
Oh, didn't see this. :+1: